packages/widgets/src/dns-hole/controls/component.tsx

```diff
@@ -322,14 +330,29 @@ const ControlsCard: React.FC<ControlsCardProps> = ({
       radius="2.5cqmin"
     >
       <Flex className="dns-hole-controls-item-container" gap="4cqmin" align="center" direction="row">
-        <Image
-          className="dns-hole-controls-item-icon"
-          src={integrationDefs[data.integration.kind].iconUrl}
-          w="20cqmin"
-          h="20cqmin"
-          fit="contain"
-          style={{ filter: !isConnected ? "grayscale(100%)" : undefined }}
-        />
+        {hasIconColor ? (
+          <ImageColored
+            src={iconUrl}
+            color="iconColor"
+            alt={data.integration.name}
+            styles={{
+              height: "20cqmin",
+              width: "20cqmin",
+              filter: !isConnected ? "grayscale(100%)" : undefined,
+            }}
+          />
+        ) : (
+          <Image
+            className="dns-hole-controls-item-icon"
+            src={integrationDefs[data.integration.kind].iconUrl}
+            w="20cqmin"
+            h="20cqmin"
+            fit="contain"
+            style={{
+              filter: !isConnected ? "grayscale(100%)" : undefined,
+            }}
+          />
+        )}
```

Comment:
It would be nice as well if you could put this in a component in the up package. Like 

```tsx
<MaskedOrNormalImage
	color="iconColor"
	src={iconUrl}
 	alt={data.integration.name}
	className="dns-hole-controls-item-icon"
	fit="contain" // Add new property to MaskedImage to pass to maskSize
        style={{
		height: "20cqmin",
		width: "20cqmin",
              	filter: !isConnected ? "grayscale(100%)" : undefined,
	}} />
```